task4/iot/apps/device/lib/device/scanner.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Device.Scanner do
+  use GenServer
+
+  def start_link(options \\ []) do
+    GenServer.start_link(__MODULE__, :ok, [options])
+  end
+    
+  def init(:ok) do
+    :dnssd.start
+    :dnssd.browse("_http._tcp")
+    {:ok, []}
+  end
+
+  def handle_info(msg, state) do
+	  {:dnssd, reference, {_, change,{name, type , domain}}} = msg
```

Comment:
This line mixes tabs and spaces for indentation. ["No tabs, just spaces" is what we want](https://github.com/christopheradams/elixir_style_guide#source-code-layout).